### PR TITLE
github-status: initial script to fetch and set github pr status

### PR DESCRIPTION
**Pull request by jdsn**
The tool to interact with the github api, needed for the mkcloud self-gating.

### CODE REVIEW

File: scripts/github-status/github-status.rb [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+ghc=GHClientHandler.new()
+
+case options[:action]
+  when 'list-unseen-prs'
+    ghc.show_unseen_pull_requests
+  when 'list-rebuild-prs'
+    ghc.show_rebuild_pull_requests
+  when 'list-forcerebuild-prs'
+    ghc.show_forcerebuild_pull_requests
```
Comment (bmwiedemann) [inline scripts/github-status/github-status.rb]: plenty duplication in this area. Would be nicer, if you just have a list of allowed params and auto-map them to method calls... or you just allow calling any method with a certain prefix, e.g. 'list-unseen-prs' could map to the cmd_list_unseen_prs method.

File: scripts/github-status/github-status.rb [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+  def is_latest_sha?(pr, sha)
+    new_sha = get_pr_latest_sha(pr)
+    new_sha == sha
+  end
+
+  def get_all_pull_requests(state, status=[])
+    pulls = @client.pull_requests(@repository, :state => state)
+    pulls.select! do |p|
+      status.include?( get_sha_status(p.head.sha) )
```
Comment (tboerger) [inline scripts/github-status/github-status.rb]: Write simply ```status.include? get_sha_status(p.head.sha)```

File: scripts/github-status/github-status.rb [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    end
+    pulls
+  end
+
+  def print_pr_sha_info(pull)
+    if pull.is_a?(Array)
```
Comment (tboerger) [inline scripts/github-status/github-status.rb]: For such simple cases you can write ```if pull.is_a? Array```

File: scripts/github-status/github-status.rb [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        print_pr_sha_info(p)
+      end
+    else
+      puts "#{pull.number}:#{pull.head.sha}"
+    end
+  end
+
+  def show_unseen_pull_requests
+    print_pr_sha_info(
+      get_all_pull_requests('open', ['']) )
```
Comment (tboerger) [inline scripts/github-status/github-status.rb]: Skip the space between ```) )```

**Comment by tboerger** [comment]
Some NITPICKing but otherwise LGTM

File: scripts/github-status/github-status.rb [hunk 5]
```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/ruby
+
+require 'octokit'
+require "optparse"
```
Comment (tboerger) [inline scripts/github-status/github-status.rb]: Still single and double quote mixing :)

File: scripts/github-status/github-status.rb [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+  def is_latest_sha?(pr, sha)
+    get_pr_latest_sha(pr) == sha
+  end
+
+  def get_all_pull_requests(state, status = [])
+    pulls = @client.pull_requests(@repository, :state => state)
+    pulls.select! do |p|
+      status.include? get_sha_status(p.head.sha)
+    end
+    pulls
```
Comment (tboerger) [inline scripts/github-status/github-status.rb]: You can skip that and do only a ```select``` above.

**Comment by MaximilianMeister** [comment]
+1

**Comment by tboerger** [comment]
LGTM

**Comment by bmwiedemann** [comment]
:+1:

**Pull request merged**